src/Sylius/Bundle/CoreBundle/Tests/Workflow/CatalogPromotionWorkflowTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\Workflow;
+
+use Sylius\Component\Core\Model\CatalogPromotion;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Workflow\StateMachine;
+
+final class CatalogPromotionWorkflowTest extends KernelTestCase
+{
+    /** @test */
+    public function it_is_a_state_machine(): void
+    {
+        $stateMachine = $this->getStateMachine();
+
+        $this->assertInstanceOf(StateMachine::class, $stateMachine);
+    }
+
+    /** @test */
+    public function it_has_states(): void
+    {
+        /** @var StateMachine $stateMachine */
+        $stateMachine = $this->getStateMachine();
+
+        $this->assertEquals([
+            'active' => 'active',
+            'inactive' => 'inactive',
+            'processing' => 'processing',
+        ], $stateMachine->getDefinition()->getPlaces());
+    }
+
+    /** @test */
+    public function it_has_activate_transition(): void
```

Comment:
I'm not sure about the values of these tests. Perhaps we could just dump the whole config and assert it? Or even better dump whole config and assert it with winzou state machine (or at least compare parts of config)

Reply:
I improved a lot this part with the both configurations. I hope you'll like it.

Reply:
Tests like `it_can_pass_from_processing_to_active` look good, but the ones in the shape of `it_has_activate_transition` are too fragile, in my opinion. Let's keep only ones in which we are setting object and checking the behaviour of the state machine instead of its configuration